From 2384e4cd63ea8cb52af6af7c7cad54ead8d78543 Mon Sep 17 00:00:00 2001
From: Ludovic Rousseau <ludovic.rousseau@free.fr>
Date: Fri, 12 Apr 2024 21:20:14 +0200
Subject: [PATCH] libpcscspy.c: do not use pcsc_stringify_error()

Fix the library loading by dlopen(), for example with OpenSC.

$ LIBPCSCLITE_DELEGATE=/lib/x86_64-linux-gnu/libpcscspy.so.0 opensc-tool -a
loading "/lib/x86_64-linux-gnu/libpcscspy.so.0" failed: /lib/x86_64-linux-gnu/libpcscspy.so.0: undefined symbol: pcsc_stringify_error
No smart card readers found.
Failed to connect to reader: No readers found
---
 src/spy/libpcscspy.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/spy/libpcscspy.c b/src/spy/libpcscspy.c
index 9877c4e9..80113529 100644
--- a/src/spy/libpcscspy.c
+++ b/src/spy/libpcscspy.c
@@ -205,8 +205,8 @@ static void spy_quit(const char *fname, LONG rv)
 	struct timeval profile_time;
 
 	gettimeofday(&profile_time, NULL);
-	spy_line("<|%ld|%ld|%s|%s|0x%08lX", profile_time.tv_sec,
-		profile_time.tv_usec, fname, pcsc_stringify_error(rv), rv);
+	spy_line("<|%ld|%ld|%s|0x%08lX", profile_time.tv_sec,
+		profile_time.tv_usec, fname, rv);
 }
 
 #define Enter() spy_enter(__FUNCTION__)
-- 
GitLab

